models/src/main/protobuf/CasperMessage.proto

```diff
@@ -246,7 +246,8 @@ message BlockMessage {
   bytes                  sig            = 7; // signature generated by signing `hash(hash(justification) concat blockHash)`.
   string                 sigAlgorithm   = 8; // name of the algorithm used to sign
   string                 shardId        = 9; // identifier of the shard where the block was created
-  bytes extraBytes = 10;
+  bool                   invalid        = 10; // whether the block was marked as invalid
```

Comment:
I'm worried about having the invalid slot as part of the BlockMessage. The BlockMessage should be immutable and the blockHash should be the hash of all the contents. It should never be modified after creation, and no validator would ever produce a block with invalid = True from the beginning (that would be slashing themselves).

Reply:
I agree that having `invalid` as a part of BlockMessage is a bad idea. I have removed it from the BlockMessage proto class but left it in the BlockMetadata proto class. Now when you are inserting a block into DAG storage you have to specify whether the block is invalid or not (i.e. I have added a new `invalid: Boolean` parameter to `BLockDagStorage[F].insert` method).